dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataOwnersManager.kt

```diff
@@ -0,0 +1,171 @@
+package org.dataland.datalandbackend.services
+
+import org.dataland.datalandbackend.entities.CompanyDataOwnersEntity
+import org.dataland.datalandbackend.repositories.DataOwnerRepository
+import org.dataland.datalandbackend.repositories.StoredCompanyRepository
+import org.dataland.datalandbackendutils.exceptions.InsufficientRightsApiException
+import org.dataland.datalandbackendutils.exceptions.InvalidInputApiException
+import org.dataland.datalandbackendutils.exceptions.ResourceNotFoundApiException
+import org.dataland.keycloakAdapter.auth.DatalandAuthentication
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import kotlin.jvm.optionals.getOrElse
+
+/**
+ * Implementation of a (company) data ownership manager for Dataland
+ * @param dataOwnerRepository  JPA for data ownership relations
+ */
+@Service("DataOwnersManager")
+class DataOwnersManager(
+    @Autowired private val dataOwnerRepository: DataOwnerRepository,
+    @Autowired private val companyRepository: StoredCompanyRepository,
+) {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Method to add a data owner to a given company
+     * @param companyId the ID of the company to which the data owner is to be added
+     * @param userId the ID of the user who is to become a data owner
+     * @return an entity holding the data ownership relations for the given company
+     */
+    @Transactional
+    fun addDataOwnerToCompany(companyId: String, userId: String): CompanyDataOwnersEntity {
+        return if (companyRepository.existsById(companyId)) {
+            if (dataOwnerRepository.existsById(companyId)) {
+                val dataOwnersForCompany = dataOwnerRepository.findById(companyId).get()
+                if (dataOwnersForCompany.dataOwners.contains(userId)) {
+                    logger.info(
+                        "User with Id $userId is already data owner of company with Id $companyId.",
+                    )
+                    dataOwnersForCompany
+                } else {
```

Comment:
Do we really need to different branches to handle a first and every other new data owner? Seems at these could be handled in one branch

Reply:
do you mean from logic perspective or just the info?
In terms of logic we still would need to distinguish between that, because of the "findbyID" function in line 38.
in case everyOtherOwner we add to the dataOwners list, in the first case we create the list

Reply:
resolved after call